flatfs.go

```diff
@@ -449,6 +462,18 @@ func (fs *Datastore) doWriteOp(oper *op) error {
 	// Do the operation
 	err := fs.doOp(oper)
 
+	// Fallback retry for temporary error.
+	if err != nil && isTooManyFDError(err) {
+		for i := 0; i < 6; i++ {
+			time.Sleep(time.Duration(i+1) * RetryDelay)
+
+			err = fs.doOp(oper)
+			if err == nil || !isTooManyFDError(err) {
+				break
+			}
+		}
+	}
```

Comment:
nit: there's a lot of repeated code here. We can do the first put attempt _inside_ the loop.